apis/oauth/src/main/java/org/jclouds/oauth/v2/config/CredentialType.java

```diff
@@ -29,7 +29,10 @@
    P12_PRIVATE_KEY_CREDENTIALS,
 
    /** Contents are an ID and Secret */
-   CLIENT_CREDENTIALS_SECRET;
+   CLIENT_CREDENTIALS_SECRET,
+
+   /** Contents are an ID and PEM-encoded Private Key.  The certificate is specified as it's own property. */
+   CLIENT_CREDENTIALS_P12;
```

Comment:
Rename it to differentiate from the Google credential, which is also a P12 private key? Perhaps something like `CLIENT_CREDENTIALS_P12_AND_CERTIFICATE`, to indicate that a certificate will also be required?